Cloud Client IP Restrictions Docs (#62478)
gravitational/teleport: docs/pages/cloud-client-ip-restrictions.mdx

---
title: Teleport Cloud Client IP Restrictions

> description: Block CIDR ranges from connecting to your cloud cluster.

Reviewer: The description is a bit misleading because it describes the inverse of the feature.

The feature is an allow list, but the description sounds a lot more like a deny list.

tags:
 - faq

>  - platform-wide

Reviewer: Is there a cloud tag?

Author: Not that I know of.

---

> **Client IP Allowlist** restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks).

Author: Updated in https://github.com/gravitational/teleport/pull/62478/commits/00201d10ef5f63102278d308d5255b1e2969bf37. I went with `Client IP restrictions` as the overall feature name since a denylist could be possible in the future.

## How to enable
This feature is currently opt-in only. Please contact your account executive or [customer support](https://goteleport.com/support/) to enable client IP restrictions for your tenant.

## Adding CIDR blocks to the IP Allowlist

Log in to your Teleport Cloud account. Open the user dropdown menu on the top right of the navigation bar, and select "Help & Support," then scroll down until you see the IP Allowlist section.
Once you add a CIDR, it will take effect in 5-20 minutes and will terminate existing connections. Changes to the allowlist are recorded in the audit log.

![Web UI view showing IP Allowlist for Teleport](../img/cloud/client-ip-restrictions.png)
Managing CIDR blocks is governed by Teleport’s existing RBAC system.
A new resource type is introduced: `client_ip_restriction`.
To see the current allowlist, the user must have the  `list` verb enabled.

To update the allowlist, `create` and `update` are required.

> By default, the `editor` role can **read and write** CIDR allowlists.

Reviewer: Alternate phrasing, take it or leave it.

```suggestion
The preset `editor` role has permissions to read and write the allowlist.

You can also create custom roles granting access via the `client_ip_restriction`
resource.

```yaml
allow:
  - resources: [ client_ip_restriction ]
    verbs:
      # list is required to view the allowlist
      - list
      # create and update are required to modify the allowlist
      - create
      - update 
      
```

```

## Limitations

### Misconfiguration

> Misconfiguration can block all access to your cluster. Make sure to include your current network before saving changes. 

Reviewer: I can almost guarantee someone will lock themselves out.

Do we have anything on the roadmap to address this footgun? Most systems with a feature like this will prevent you from blocking your own IP (or at least warn you first).

---
title: Teleport Cloud Client IP Restrictions
description: Block CIDR ranges from connecting to your cloud cluster.
tags:
 - faq
 - platform-wide
---

> Client IP Allowlist restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks). This feature is currently [opt-in](#how-to-enable).

Reviewer 2: ```suggestion
**Client IP Allowlist** restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks). This feature is currently [opt-in](#how-to-enable).
```

Reviewer 2: (Product names should be bolded on first use per our style guide)

> Client IP Allowlist restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks). This feature is currently [opt-in](#how-to-enable).

Reviewer 2: ```suggestion
Client IP Allowlist restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks).
```

## How to enable
Please contact your account executive or customer support to enable client IP restrictions for your tenant.

## Adding CIDR blocks to the IP Allowlist

Log in to your Teleport Cloud account. Open the user dropdown menu on the top right of the navigation bar, and select "Help & Support," then scroll down until you see the IP Allowlist section.

> Once you add a CIDR, they will take effect in 5-20 minutes and will terminate existing connections. Changes to the allowlist are recorded in the audit log.

Reviewer 2: ```suggestion
Once you add a CIDR, it will take effect in 5-20 minutes and will terminate existing connections. Changes to the allowlist are recorded in the audit log.
```

![Web UI view showing IP Allowlist for Teleport](../img/cloud/client-ip-restrictions.png)

> 

Reviewer 2: ```suggestion
```

Managing CIDR blocks is governed by Teleport’s existing RBAC system.

> 

Reviewer 2: ```suggestion
```

A new resource type is introduced: `client_ip_restriction`.

> 

Reviewer 2: ```suggestion
```

Reviewer 2: We can just make this one paragraph, no need for spaces.

In order to see the current allowlist, the user must have the `list` verb enabled.

To update the allowlist, `create` and `update` are required.

By default, the `editor` role can **read and write** CIDR allowlists.

## Limitations

### Misconfiguration
Misconfiguration can block all access to your cluster. Make sure to include your current network before saving changes. 

### third-party service ranges
Teleport does not auto-add third-party service ranges. You must add all required CIDRs explicitly. 

> ### Network Security

Reviewer 2: ```suggestion
### Network security
```

The allowlist applies to Teleport Cloud access; it does not replace your organization’s network/firewall policies.

> ### Sync Time

Reviewer 2: ```suggestion
### Sync time
```

The Client IP Restriction allowlist may take up to 20 minutes before it is fully synced. 

## FAQ

### How many CIDRs can you configure?

By default, up to 256 CIDR blocks can be configured. Please contact your account executive or customer support to increase the limit. 

### Do you support a denylist?

> Teleport Cloud Client IP Restrictions doesn't currently support a denylist. 

Reviewer 2: Not a proper noun, unless 'Client IP Restrictions' is the name of one of our products/services? If so, ignore this one! (But this is ok for the title field per our style guide.)

---
title: Teleport Cloud Client IP Restrictions
description: Restrict access to your cloud cluster with a configurable allowlist.
tags:
 - faq
 - platform-wide
---

> **Client IP Restrictions** restricts access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks).

Reviewer: ```suggestion
**Client IP Restrictions** restrict access to your Teleport Cloud cluster, allowing traffic only from the specified network ranges (CIDR blocks).
```
